ENT-3284 postman collection and readme

binodpant opened the pull request:
**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `make static` has been run to update webpack bundling if any static content was updated
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [x] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [x] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [x] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

taliaedX commented on integrated_channels/canvas/dev_docs/README.md:
```diff
@@ -0,0 +1,18 @@
+Setup a postman environment and setup these vars
+
+* canvas_url: usually just http://localhost:3000
+* access_token: once you have it
```
Is this the bearer token or something else?

binodpant replied:
yup updated doc

taliaedX commented on integrated_channels/canvas/dev_docs/postman_collections/Canvas.postman_collection.json:
```diff
@@ -0,0 +1,445 @@
+						},
+						{
+							"key": "client_secret",
+							"value": "bSxFhGIBmqluIMe7fIc0S6fNU75WtD7sB7r6T8euVwoYJ6eeNZIU0cfBS4VwjHwJ",
```
Is there a way to actually make the template "blank"? ie it has the key client_secret, but the template is not storing its value.
I know these are just for development, but it still seems odd to commit these values.

binodpant replied:
yah I do not know of a good way to do that, I think I could set the environment to none or sth in postman then export these.. but for this round I can just clear these out of the json

binodpant replied:
actually I can and should set these to the variable instead of the value

binodpant replied:
updated to use vars that can be sett in postman, instead of raw values

taliaedX commented on integrated_channels/canvas/dev_docs/postman_collections/Canvas.postman_collection.json:
```diff
@@ -0,0 +1,445 @@
+					"formdata": [
+						{
+							"key": "course[image_url]",
+							"value": "lt5H1qDZfNo9aglt7jb9fLCLdRb1cEZNMdD3VZjORE3o82jyV1LAxNVI0up5fnur",
```
why is the image url the same as the token? lol.
Given you have access_token as a var, I'd replace all spots with the token's value with the variable `{{access_token}}`

binodpant replied:
haha fixed it

mattdrayer commented:
Just a heads up that we have a private repository for Postman collections, if you want to use it: https://github.com/edx/edx-postman-config

georgebabey commented on integrated_channels/canvas/dev_docs/README.md:
```diff
@@ -0,0 +1,18 @@
+
+Just to help local dev and testing with Canvas
+
+Firstt ensure you have canvas setup as per
```
nit: `First`

binodpant commented:
> Just a heads up that we have a private repository for Postman collections, if you want to use it: https://github.com/edx/edx-postman-config

Thanks Matt. I was hoping to keep this local to this code but I also see the value of the private repo. For this instance since we only committed non prod or non sensitive values, does this still work?

mattdrayer commented:
I think it's really up to you all -- see the benefit of keeping the collection close to the code, however there are other people at edX who use Postman to connect to these systems (think: ECS, Sales Engineers) and so it might be easier for them to find collections if they are all in one place.  Also it's very possible that over time collections can/will grow to include more than what is implemented in the code.

At the end of the day we're not really talking about a lot of people requiring access to Postman collections, and so with proper documentation/pointers we can probably avoid confusion about where things live.

binodpant commented:
believe I addressed the feedback, except we can reconsider if moving this content to the other repo in the future makes sense we can do that. otherwise all done; version, changelog also updated

taliaedX commented on integrated_channels/canvas/dev_docs/postman_collections/Canvas.postman_collection.json:
```diff
@@ -58,17 +58,17 @@
 						{
 							"key": "code",
-							"value": "b77ce731852e1691c69eac9ac5bf8d62bdf8ae5f59c521726e37e0aa87ab49a6561b615146ff918e0bd79556c055a81f9fce589e35c5bf98eaafa88f703939bf",
+							"value": "{{code}}",
```
Just a heads up that this var isn't mentioned in the readme.

binodpant replied:
nice catch got it!

taliaedX commented on integrated_channels/canvas/dev_docs/postman_collections/Canvas.postman_collection.json:
```diff
@@ -0,0 +1,445 @@
+				"method": "GET",
+				"header": [],
+				"url": {
+					"raw": "{{canvas_url}}/login/oauth2/auth?client_id=10000000000002&response_type=code&state=edx-canvas-user-state&redirect_uri=https://example.com/oauth_complete",
```
shouldn't `client_id=10000000000002` be `client_id={{client_id}}` here?

binodpant replied:
yeah we can use that var too, I had been plugging in different values when testing multiple keys but better as a var, changing

taliaedX commented on integrated_channels/canvas/dev_docs/postman_collections/Canvas.postman_collection.json:
```diff
@@ -0,0 +1,445 @@
+				"method": "GET",
+				"header": [],
+				"url": {
+					"raw": ""
```
Am I wrong or is this basically just a blank request template?

binodpant replied:
hmm will try re exporting it is not empty in postmane for me

taliaedX commented on integrated_channels/canvas/dev_docs/postman_collections/Canvas.postman_collection.json:
```diff
@@ -0,0 +1,445 @@
+				"method": "GET",
+				"header": [],
+				"url": {
+					"raw": "{{canvas_url}}/api/v1/courses",
```
so according to the doc this endpoint gets courses for the "current user". I'm not sure how well this will work if the given user is not actually enrolled in any of these courses. I wonder if we should go by account id here, ie `{{canvas_url}}/api/v1/accounts/:account_id/courses/` but what has been your experience with courses you aren't specifically enrolled in?
